### ISSUE TITLE
TZID prefix added to UTC datetimes

### ISSUE THREAD

**Issue by hprange** [ical4j/ical4j#663]
**Describe the bug**
I've encountered an issue while using the latest release candidate of ical4j. When I create a `VEvent` with dates set as `ZonedDateTime` instances using `ZoneOffset.UTC`, the dates are prefixed with `TZID`.

**To Reproduce**
Here's a snippet of code that generates a `VEvent` for a one-hour meeting, with the start and end dates set in UTC:

```java
var startDate = ZonedDateTime.of(2023, 11, 15, 8, 30, 0, 0, ZoneOffset.UTC);
var endDate = startDate.plusHours(1);

System.out.print(new VEvent(startDate, endDate, "Metting"));
```

The output I'm getting is:

```
startDate ==> 2023-11-15T08:30Z
endDate ==> 2023-11-15T09:30Z

BEGIN:VEVENT
DTSTAMP:20231118T154444Z
DTSTART;TZID=Z:20231115T083000Z
DTEND;TZID=Z:20231115T093000Z
SUMMARY:Metting
END:VEVENT
```

**Expected behavior**
According to [RFC5545](https://datatracker.ietf.org/doc/html/rfc5545#section-3.2.19):

> The "TZID" property parameter MUST NOT be applied to DATE properties and DATE-TIME or TIME properties whose time values are specified in UTC.

Given this, my expectation was for the `VEvent` to be generated without the `TZID` prefix, as shown below:

```
BEGIN:VEVENT
DTSTAMP:20231118T154444Z
DTSTART:20231115T083000Z
DTEND:20231115T093000Z
SUMMARY:Metting
END:VEVENT
```

**Environment**
 - OS: macOS 14.1.1
 - Java: 15.0.2
 - iCal4j: 4.0.0-rc1

**Comment by benfortuna** [comment]
Thanks for reporting, I will look into a fix for this. 

Note that for now a workaround is to use Instant rather than ZonedDateTime for UTC dates. For example:

`var startDate = ZonedDateTime.of(2023, 11, 15, 8, 30, 0, 0, ZoneOffset.UTC).toInstant();`

**Comment by benfortuna** [comment]
A fix has been applied to remove TZID params for UTC DATE-TIME values. This is now available in 4.0.0-rc2.

**Comment by jannis3005** [comment]
This seems to still be a thing in `Version 4.1.0`. We're using ZonedDateTimes with ZoneID UTC. The Output is TZID=UTC without a trailing Z. If we change the ZoneID to Z manually, the Trailing Z is added, but it still adds TZID=Z.

We're using the `.toInstant()` Workaround for now, as that produces the desired result
